dnf-behave-tests/createrepo_c/steps/lib/file.py

```diff
@@ -95,3 +95,21 @@ def decompress_file_by_extension_to_dir(compressed_filepath, dest_dir):
 
     open(dst, "wb").write(content)
     return dst
+
+
+def create_compressed_file_with_contents(filename, compression, contents, encoding="utf-8"):
+    fullname = filename + get_compression_suffix(compression)
+    if os.path.exists(fullname):
+        raise ValueError("File: " + fullname + " already exists")
+
+    if compression == "gz":
+        with gzip.open(fullname, 'wt') as f:
+            f.write(contents)
+    elif compression == "xz":
+        with lzma.open(fullname, 'wt') as f:
+            f.write(contents)
+    elif compression == "bz2":
+        with bz2.open(fullname, 'wt') as f:
+            f.write(contents)
+    else:
+        raise ValueError("Unknown compression type: " + compression)
```

Comment:
Is there any reason why not put the body of this function straight to the step in `dnf-behave-tests/createrepo_c/steps/cmd.py`? It doesn't seem probable this function will be useful in another place and then the indirection just makes it harder to read the code and has no benefit?

Reply:
I didn't want to import all the compression libraries again. But I can move it if you don't think is worth it?

Reply:
Yeah, ok... probably reason enough to leave it this way...